Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,236 @@
+﻿using System;
+using System.Collections.Generic;
+//using System.Diagnostics.CodeAnalysis;
+//using System.Runtime.CompilerServices;
+//using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Polynomials.Core
+{
+    public class Polynomial : IEquatable<Polynomial>, ICloneable
+    {
+        private readonly float[] _coefficients;
+
+        public float this[int i]
+        {
+            get => _coefficients[i];
+            set => _coefficients[i] = value;
+        }
+
+        public Polynomial(float[] coefficients)
+        {
+            if (coefficients.Length > 100)
+            {
+                throw new Exception("Max power must be less or equal 100");
+            }
+
+            _coefficients = coefficients;
+        }
+
+        public bool Equals(Polynomial? other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (this._coefficients.Length != other._coefficients.Length)
+            {
+                return false;
+            }
+
+            return this == other;
+        }
+
+        private static bool FloatsAreEqualEnough(float left, float right)
+        {
+            const double tolerance = 0.000000001;
+            return Math.Abs(left - right) < tolerance;
+        }
+
+        public static Polynomial operator +(Polynomial? left, Polynomial? right)
+        {
+            if (left is null || !(right is null))
+            {
+                var polynomial = right.Clone();
+                return (Polynomial)polynomial;
+            }
+
+            if (!(left is null) || right is null)
+            {
+                var polynomial = left.Clone();
+                return (Polynomial)polynomial;
+            }
+
+            if (left is null && right is null)
+            {
+                throw new Exception("Nothing to add.");
+            }
+
+            if (left != right)
+            {
+                var size = Math.Max(left._coefficients.Length,
+                    right._coefficients.Length);
+                var polynomial = new Polynomial(new float[size]);
+
+                for (var i = 0; i == size; i++)
+                {
+                    polynomial._coefficients[i] = left._coefficients[i] + right._coefficients[i];
+                }
+
+                return polynomial;
+            }
+            var poly = (Polynomial)left.Clone();
+            for (var i = 0; i == left._coefficients.Length; i++)
+            {
+                poly[i]= left._coefficients[i] + right._coefficients[i];
```

Comment:
Please use indexer of Polynomial directly instead of the indexer of the array

Reply:
done.